src/IssueViz/Editor/LocationNavigator.cs

```diff
@@ -47,33 +49,43 @@ internal LocationNavigator(IDocumentNavigator documentNavigator, IIssueSpanCalcu
             this.logger = logger;
         }
 
-        public bool TryNavigate(IAnalysisIssueLocation location)
+        public bool TryNavigate(IAnalysisIssueLocationVisualization locationVisualization)
         {
-            var locationFilePath = location?.FilePath;
-
-            if (string.IsNullOrEmpty(locationFilePath))
+            if (locationVisualization == null)
             {
-                return false;
+                throw new ArgumentNullException(nameof(locationVisualization));
             }
 
             try
             {
-                var textView = documentNavigator.Open(locationFilePath);
-                var locationSpan = spanCalculator.CalculateSpan(location, textView.TextBuffer.CurrentSnapshot);
+                var textView = documentNavigator.Open(locationVisualization.CurrentFilePath);
+                var locationSpan = GetLocationSpan(locationVisualization, textView);
 
-                if (locationSpan.HasValue)
+                if (!locationSpan.IsEmpty)
                 {
-                    documentNavigator.Navigate(textView, locationSpan.Value);
+                    documentNavigator.Navigate(textView, locationSpan);
 
                     return true;
                 }
             }
             catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                logger.WriteLine(Resources.ERR_OpenDocumentException, locationFilePath, ex.Message);
+                logger.WriteLine(Resources.ERR_OpenDocumentException, locationVisualization.CurrentFilePath, ex.Message);
             }
 
             return false;
         }
+
+        private SnapshotSpan GetLocationSpan(IAnalysisIssueLocationVisualization locationVisualization, ITextView textView)
+        {
+            var shouldCalculateSpan = !locationVisualization.Span.HasValue || locationVisualization.Span.Value.IsEmpty;
```

Comment:
An empty span means we've previously been unable to calculate the span correctly e.g. because the file has been edited. Why would we try to recalculate the span from the original location in this case?

Reply:
You're right; I'll change it.